app/controllers/users_controller.rb

```diff
@@ -1,11 +1,15 @@
 # Users controller to manage show and update actions.
 class UsersController < ApplicationController
   skip_before_action :verify_authenticity_token, only: [:update]
-  respond_to :json
 
   def show
-    user = User.find(params[:id])
-    render json: user
+    user = User.find_by(id: params[:id])
```

Comment:
Use User.find instead of User.find_by, this way it will throw a Not found error, in ApplicationController rescue this error and return a not_found code